tweets.py

```diff
@@ -59,30 +67,85 @@ def get_observations(place):
 # however, the observations aren't necessarily spaced equally throughout
 # the day. do something more sophisticated later!
 def get_daily_temp(place):
-    observations = get_observations(place)
-
-    temps = [temp for (timestamp, temp) in observations]
+    temps = [temp for (timestamp, temp) in observations if temp]
     average = list_average(temps)
     average_fahrenheit = average * 1.8 + 32
 
     return average_fahrenheit
 
-# This currently calculates the average temperature for the current day
-def get_historical_temps(place):
-    historical_data_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
+def get_historical_temps(place, local_time):
+    place_id = place['USAF'] + "-" + place['WBAN']
+
+    current_day_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
         data_url=DATA_URL,
-        id=place['USAF'] + "-" + place['WBAN'],
+        id=place_id,
         month='{:02}'.format(END_TIME.month),
         day='{:02}'.format(END_TIME.day)
     )
-    response = requests.get(historical_data_url)
-    lines = (line.decode('utf-8') for line in response.iter_lines())
-    reader = csv.reader(lines)
-    next(reader, None)  # skip the headers
+
+    current_day = get_reader(current_day_url)
+
+    # we want the last 24 hours of weather data, ending at 6pm today local time
+    tz_offset = int(local_time.utcoffset().total_seconds() / 3600)
+    additional_hours_needed = -6 - tz_offset
+
+    if additional_hours_needed < 0:
+        # we need to grab the previous day's sheet as well and take the last
+        # additional_hours_needed hours for each year
+        previous_day = END_TIME - Timedelta(days=1)
+        previous_day_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
+            data_url=DATA_URL,
+            id=place_id,
+            month='{:02}'.format(previous_day.month),
+            day='{:02}'.format(previous_day.day)
+        )
+
```

Comment:
I didn't check this logic. I think it would be cleaner to just check whether `START_TIME.date() == END_TIME.date()` to see if an additional sheet is needed.

Reply:
Can you say more? For Chicago we only need one sheet but `START_TIME.date() == END_TIME.date()` will return false because right now it looks something like this: 

`{'start': '2020-02-24T00:00:00+00:00', 'end': '2020-02-25T00:00:00+00:00'}`

We'll still need to check if the place is ahead of or behind UTC, which is what most of this logic is dealing with. This is the part I could use a check on!

Reply:
Oh yeah there's a fencepost thing happening here. So let `start_time = end_time - Timedelta(1, 'D') + Timedelta(1, 'm')` then the check does work. And it never requires looking at local time so i think it is less error prone.

I didn't follow what you said about being "ahead or behind UTC"-- does what I'm suggesting avoid that?

Reply:
Whoops I may have adjusted the wrong fencepost... I hope you know what I mean?

Reply:
I think I'm seeing what you mean. I've pushed some changes now that keep this logic in UTC, what do you think?

Reply:
Cool, I will take a look soon, maybe tonight but Tues-Thurs I don't have much free time

Reply:
Ya no hurry!